### PR TITLE
Chroma search optimization 2

### PR DESCRIPTION
**Pull request by NaderMahdi**
## Description
Perform the Chroma search algorithm just before the last md stage and only when at least one intra mode make it to this stage.
This is a replacement of PR#1039

## Related issue
#1047 

## Performance (from PR1039)
* Bdrate 0.02% Bdrate loss
* Speed 3.5% speed up
* Memory 1% increase

### CODE REVIEW

File: Source/Lib/Common/Codec/EbDefinitions.h [hunk 1]
```diff
@@ -34,6 +34,12 @@
 extern "C" {
 #endif
 
+#define CHROMA_SEARCH_OPT        1 // Move chroma search to be done on the best intra candidate survived from MD stage 2
```
Comment (fouedb) [inline Source/Lib/Common/Codec/EbDefinitions.h]: // Chroma search to be performed only for best intra candidate that survives from MD stage 2.

File: Source/Lib/Common/Codec/EbDefinitions.h [hunk 2]
```diff
@@ -34,6 +34,12 @@
...
+#if CHROMA_SEARCH_OPT
+#define INFR_OPT                 1 // Lossless: Infrastructure work to allow the protability of the chroma search
+#define MOVE_OPT                 1 // Semi-lossless: Move the intra search to just before last md_stage
```
Comment (fouedb) [inline Source/Lib/Common/Codec/EbDefinitions.h]: Please clarify.

File: Source/Lib/Common/Codec/EbDefinitions.h [hunk 3]
```diff
@@ -34,6 +34,12 @@
...
+#define COMP_OPT                 1 // Speed optimisation by restricting the search only when at least 1 intra candidate survives to the last md_stage
```
Comment (fouedb) [inline Source/Lib/Common/Codec/EbDefinitions.h]: Speed optimization  - Restrict the search only to the case when at least 1 intra candidate survives to the last md_stage. 

Clarify what "Restrict the search" means.

**Comment by mafonsoN** [comment]
Can you put the performance impact of these changes here again?
Did you apply the clang format to these changes?

**Comment by hassount** [comment]
> Did you apply the clang format to these changes?

The clang format seems to be creating some funky lines, there's someone working on fixing this, then we can apply it to all files again to fix everything.

File: Source/Lib/Common/Codec/EbDefinitions.h (hunk 1 above)
Comment (NaderMahdi) [reply Source/Lib/Common/Codec/EbDefinitions.h]: These are Macros that will be removed eventually.

File: Source/Lib/Common/Codec/EbDefinitions.h (hunk 2 above)
Comment (NaderMahdi) [reply Source/Lib/Common/Codec/EbDefinitions.h]: These are Macros that will be removed eventually.

File: Source/Lib/Common/Codec/EbDefinitions.h (hunk 3 above)
Comment (NaderMahdi) [reply Source/Lib/Common/Codec/EbDefinitions.h]: These are Macros that will be removed eventually.

**Pull request merged**